File: test/aidbox_sdk/snapshots/dotnethl7-fhir-r4-core/Coverage.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class Coverage : DomainResource, IResource
{
    public Base.ResourceReference? PolicyHolder { get; set; }
    public required Base.ResourceReference Beneficiary { get; set; }
    public Base.ResourceReference[]? Contract { get; set; }
    public Base.CodeableConcept? Relationship { get; set; }
    public Base.CodeableConcept? Type { get; set; }
    public Coverage_CostToBeneficiary[]? CostToBeneficiary { get; set; }
    public bool? Subrogation { get; set; }
    public Base.ResourceReference? Subscriber { get; set; }
    public required Base.ResourceReference[] Payor { get; set; }
    public required string Status { get; set; }
    public Coverage_Class[]? Class_ { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public string? Order { get; set; }
    public string? Network { get; set; }
    public Base.Period? Period { get; set; }
    public string? Dependent { get; set; }
    public string? SubscriberId { get; set; }

    public class Coverage_CostToBeneficiary_Exception : BackboneElement
    {
        public required Base.CodeableConcept Type { get; set; }
        public Base.Period? Period { get; set; }
    }

    public class Coverage_CostToBeneficiary : BackboneElement
    {
        public Base.CodeableConcept? Type { get; set; }
        public object? Value    
        {
            get
            {
                if (ValueMoney is not null)
                {
                    return ValueMoney;
                }
        
                if (ValueQuantity is not null)
                {
                    return ValueQuantity;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(Base.Money))
                {
                    ValueMoney = (Base.Money)value;
                    return;
                }
        
                if (value?.GetType() == typeof(Base.Quantity))
                {
                    ValueQuantity = (Base.Quantity)value;
                    return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public Coverage_CostToBeneficiary_Exception[]? Exception { get; set; }
        public Base.Money? ValueMoney { get; set; }
        public Base.Quantity? ValueQuantity { get; set; }
    }

    public class Coverage_Class : BackboneElement
    {
        public string? Name { get; set; }
        public required Base.CodeableConcept Type { get; set; }
        public required string Value { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/Coverage.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Period
from base import CodeableConcept
from base import Quantity
from base import DomainResource
from base import Money
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class CoverageCostToBeneficiaryException(BackboneElement):
    type: CodeableConcept
    period: Optional[Period] = None

@dataclass(kw_only=True)
class CoverageCostToBeneficiary(BackboneElement):
    type: Optional[CodeableConcept] = None
    exception: Optional[List[CoverageCostToBeneficiaryException]] = field(default_factory=list)
    value_money: Optional[Money] = None
    value_quantity: Optional[Quantity] = None

@dataclass(kw_only=True)
class CoverageClass(BackboneElement):
    name: Optional[str] = None
    type: CodeableConcept
    value: str

@dataclass(kw_only=True)
class Coverage(DomainResource):
    policy_holder: Optional[Reference] = None
    beneficiary: Reference
    contract: Optional[List[Reference]] = field(default_factory=list)
    relationship: Optional[CodeableConcept] = None
    type: Optional[CodeableConcept] = None
    cost_to_beneficiary: Optional[List[CoverageCostToBeneficiary]] = field(default_factory=list)
    subrogation: Optional[bool] = None
    subscriber: Optional[Reference] = None
    payor: list[Reference] = field(default_factory=list)
    status: str
    class_: Optional[List[CoverageClass]] = field(default_factory=list)
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    order: Optional[int] = None
    network: Optional[str] = None
    period: Optional[Period] = None
    dependent: Optional[str] = None
    subscriber_id: Optional[str] = None